### ISSUE TITLE
Infinite cycle when computer recovers from suspended state

### ISSUE THREAD

**Issue by gy-lehel** [tieto/sipe#42]
```
ii  pidgin                                                1:3.0.0~collab-0.20610+20160321 amd64                           graphical multi-protocol instant messaging client for X
ii  pidgin-data                                           1:3.0.0~collab-0.20610+20160321 all                             multi-protocol instant messaging client - data files
ii  pidgin-dbg                                            1:3.0.0~collab-0.20610+20160321 amd64                           Debugging symbols for Pidgin
un  pidgin-facebookchat                                   <none>                          <none>                          (no description available)
ii  pidgin-gnome-keyring                                  2.0~trusty-1                    amd64                           integrates pidgin (and libpurple) with the system keyring
ii  pidgin-otr                                            4.0.0-2                         amd64                           Off-the-Record Messaging plugin for Pidgin
rc  pidgin-ppa                                            0.0.9                           all                             Pidgin PPA
ii  pidgin-sipe                                           1.20.1+collab-201603211059+3181 amd64                           Pidgin plugin for MS Office Communicator and MS Lync
ii  pidgin-sipe-dbg                                       1.20.1+collab-201603211059+3181 amd64                           Pidgin plugin for MS Office Communicator and MS Lync (debug symbols)
```

```
(gdb) where
#0  0x00007f8f0659e95b in __libc_send (fd=36, buf=0x55f30a3ef210, n=2093, flags=106555739) at ../sysdeps/unix/sysv/linux/x86_64/send.c:31
#1  0x00007f8ef7d9d876 in ?? () from /usr/lib/x86_64-linux-gnu/libnspr4.so
#2  0x00007f8ef02bc0bc in ?? () from /usr/lib/x86_64-linux-gnu/libssl3.so
#3  0x00007f8ef02c118e in ?? () from /usr/lib/x86_64-linux-gnu/libssl3.so
#4  0x00007f8ef02c2590 in ?? () from /usr/lib/x86_64-linux-gnu/libssl3.so
#5  0x00007f8ef02c63b8 in ?? () from /usr/lib/x86_64-linux-gnu/libssl3.so
#6  0x00007f8ef0717295 in ssl_nss_write (gsc=0x24, data=0x55f30a3ef210, len=2093) at ssl-nss.c:615
#7  0x00007f8ee8ce6ded in transport_write (transport=transport@entry=0x55f30a1e28e0) at purple-transport.c:368
#8  0x00007f8ee8ce7308 in sipe_backend_transport_flush (conn=0x55f30a1e28e0) at purple-transport.c:430
#9  0x00007f8ee8cbb090 in sipe_core_deallocate (sipe_public=0x55f30a551e00) at sipe-core.c:435
#10 0x00007f8ee8ca97ee in sipe_purple_close (gc=0x7f8e38003ae0) at purple-plugin-common.c:519
#11 0x00007f8f013ca046 in purple_connection_finalize (object=0x7f8e38003ae0) at connection.c:789
#12 0x00007f8f01a1ddba in g_object_unref () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
#13 0x00007f8f013aab18 in purple_account_disconnect (account=0x55f308adea80) at account.c:385
#14 0x00007f8f013c8df5 in purple_connection_disconnect_cb (data=0x55f308adea80) at connection.c:456
#15 0x00007f8f067f6703 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#16 0x00007f8f067f5ce5 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#17 0x00007f8f067f6048 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#18 0x00007f8f067f630a in g_main_loop_run () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#19 0x00007f8f02a3ce25 in gtk_main () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
#20 0x00007f8f06b7cfc5 in pidgin_start (argc=1, argv=0x7ffddeefd118) at libpidgin.c:850
#21 0x00007f8f061ebec5 in __libc_start_main (main=0x55f307fdb960 <main>, argc=1, argv=0x7ffddeefd118, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffddeefd108) at libc-start.c:287
#22 0x000055f307fdb9de in _start ()
```

**Comment by gy-lehel** [comment]
```
#0  __errno_location () at ../csu/errno-loc.c:26
No locals.
#1  0x00007f40c593b0d5 in transport_write (transport=transport@entry=0x55c58bba4960) at purple-transport.c:375
        written = <optimized out>
        max_write = 6920
#2  0x00007f40c593b5a8 in sipe_backend_transport_flush (conn=0x55c58bba4960) at purple-transport.c:430
        transport = 0x55c58bba4960
#3  0x00007f40c590f350 in sipe_core_deallocate (sipe_public=0x55c58cf57d40) at sipe-core.c:435
        sipe_private = 0x55c58cf57d40
#4  0x00007f40c58fd74e in sipe_purple_close (gc=0x55c58cf5a470) at purple-plugin-common.c:519
        purple_private = 0x55c58c7eeec0
        sipe_public = <optimized out>
#5  0x00007f40dd3714c6 in purple_connection_finalize (object=0x55c58cf5a470) at connection.c:789
        gc = 0x55c58cf5a470
        priv = 0x55c58cf5a410
        account = 0x55c58bbf8270
        buddies = <optimized out>
        remove = 1
#6  0x00007f40dd9c5dba in g_object_unref () from /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0
No symbol table info available.
#7  0x00007f40dd355938 in purple_account_disconnect (account=0x55c58bbf8270) at account.c:385
        gc = <optimized out>
        priv = 0x55c58bbf81c0
        username = <optimized out>
        __FUNCTION__ = "purple_account_disconnect"
#8  0x00007f40dd370275 in purple_connection_disconnect_cb (data=0x55c58bbf8270) at connection.c:456
        account = 0x55c58bbf8270
        gc = 0x55c58cf5a470
        priv = <optimized out>
#9  0x00007f40e2b15703 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#10 0x00007f40e2b14ce5 in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#11 0x00007f40e2b15048 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#12 0x00007f40e2b1530a in g_main_loop_run () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
No symbol table info available.
#13 0x00007f40ded57e25 in gtk_main () from /usr/lib/x86_64-linux-gnu/libgtk-3.so.0
No symbol table info available.
#14 0x00007f40e2e9b435 in pidgin_start (argc=1, argv=0x7ffce3c830c8) at libpidgin.c:850
        opt_force_online = 0
        opt_help = <optimized out>
        opt_login = 0
        opt_nologin = 0
        opt_version = 0
        opt_si = 1
        opt_config_dir_arg = 0x0
        opt_login_arg = <optimized out>
        opt_session_arg = 0x0
        search_path = <optimized out>
        provider = <optimized out>
        screen = <optimized out>
        accounts = <optimized out>
        sig_indx = 1
        sigset = {__val = {82950, 0 <repeats 15 times>}}
        errmsg = "\000\017\310\343\374\177\000\000\070\333*\343@\177\000\000\300\060\310\343\374\177\000\000@\003\000\000\000\000\000\000\177ELF\002\001\001\000\000\000\000\000\000\000\000\000\003\000>\000\001\000\000\000\240\t\000\000\000\000\000\000@\000\000\000\000\000\000\000pA\000\000\000\000\000\000\000\000\000\000@\000\070\000\a\000@\000\032\000\031\000\001\000\000\000\005", '\000' <repeats 27 times>, "\\0\000\000\000\000\000\000\\0\000\000\000\000\000\000\000\000 \000\000\000\000\000\001\000\000\000\006\000\000\000\350=\000\000\000\000\000\000\350= \000\000\000\000\000\350= \000\000\000\000\000\200\002\000\000\000\000\000\000\210\002\000\000\000\000\000\000\000\000"...
        signal_channel = <optimized out>
        signal_status = <optimized out>
        signal_channel_watcher = 1
        error = 0x7f40c8001da0
        segfault_message_tmp = <optimized out>
        opt = <optimized out>
        gui_check = <optimized out>
        debug_enabled = <optimized out>
        debug_colored = <optimized out>
        active_accounts = <optimized out>
        st = {st_dev = 64513, st_ino = 131924, st_nlink = 2, st_mode = 16832, st_uid = 79260, st_gid = 12565, __pad0 = 0, st_rdev = 0, st_size = 4096, st_blksize = 4096, st_blocks = 8, st_atim = {tv_sec = 1445415107, 
            tv_nsec = 279910880}, st_mtim = {tv_sec = 1445415107, tv_nsec = 279910880}, st_ctim = {tv_sec = 1445415107, tv_nsec = 279910880}, __glibc_reserved = {0, 0, 0}}
        long_options = {{name = 0x7f40e2eb0d28 "config", has_arg = 1, flag = 0x0, val = 99}, {name = 0x7f40e2e9e1c8 "debug", has_arg = 2, flag = 0x0, val = 100}, {name = 0x7f40e2eb544a "force-online", has_arg = 0, flag = 0x0, val = 102}, 
          {name = 0x7f40e2e9fdd6 "help", has_arg = 0, flag = 0x0, val = 104}, {name = 0x7f40e2eb532f "login", has_arg = 2, flag = 0x0, val = 108}, {name = 0x7f40e2eb5457 "multiple", has_arg = 0, flag = 0x0, val = 109}, {
            name = 0x7f40e2eb5460 "nologin", has_arg = 0, flag = 0x0, val = 110}, {name = 0x7f40e2eb0d1e "session", has_arg = 1, flag = 0x0, val = 115}, {name = 0x7f40e2ea2ddd "version", has_arg = 0, flag = 0x0, val = 118}, {
            name = 0x7f40e2eb0d31 "display", has_arg = 1, flag = 0x0, val = 68}, {name = 0x7f40e2eac935 "sync", has_arg = 0, flag = 0x0, val = 83}, {name = 0x0, has_arg = 0, flag = 0x0, val = 0}}
#15 0x00007f40e250aec5 in __libc_start_main (main=0x55c58a29c960 <main>, argc=1, argv=0x7ffce3c830c8, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffce3c830b8) at libc-start.c:287
        result = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {0, 5638528875885417734, 94306914912693, 140724130033856, 0, 0, -5640244291729173242, -5710053594924526330}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x7ffce3c830d8, 
              0x7f40e33011c8}, data = {prev = 0x0, cleanup = 0x0, canceltype = -473419560}}}
        not_first_call = <optimized out>
#16 0x000055c58a29c9de in _start ()
```

**Comment by xhaakon** [comment]
Merged with #54.
